Code/TestProject2/CalcXUnitTests.cs

```diff
@@ -0,0 +1,143 @@
+        [Theory]
+        [InlineData(5, 2, 3)]
+        [InlineData(-5, 2, -7)]
+        [InlineData(0, 0, 0)]
+        [InlineData(-10, -2, -8)]
+        public void SubtractReturnCorrectDifferenceTest(double a, double b, double expectedDifference)
+        {
+            // Precondition: Two double numbers a and b to subtract, and the expected difference expectedDifference.
+            //Test: Check that the Sub method returns the correct difference for different input values.
+            double actualResult = calculator.Sub(a, b);
+            Assert.Equal(expectedDifference, actualResult);
+        }
+
+        [Theory]
+        [InlineData(3, 4, 12)]
+        [InlineData(-3, 4, -12)]
+        [InlineData(0, 0, 0)]
+        [InlineData(-2, -3, 6)]
+        public void MultiplyReturnCorrectProductTest(double a, double b, double expectedProduct)
+        {
+            //Precondition: Two double numbers a and b to multiply, and the expected product expectedProduct.
+            //Test: Check that Multiply method returns the correct product for different input values.
+            double actualResult = calculator.Multiply(a, b);
+            Assert.Equal(expectedProduct, actualResult);
+        }
+
+        [Theory]
+        [InlineData(20, 4, 5)]
+        [InlineData(0, -7, 0)]
+        [InlineData(-16, 4, -4)]
+        public void DivideReturnCorrectQuotient(double a, double b, double expectedQuotient)
+        {
+            //Precondition: Two double numbers a and b to divide, and the expected quotient expectedQuotient.
+            //Test: Check that the Divide method returns the correct quotient for different input values.
+            double actualResult = calculator.Divide(a, b);
+            Assert.Equal(expectedQuotient, actualResult);
+        }
+
+        [Theory]
+        [InlineData(4, 2)]
+        [InlineData(9, 3)]
+        [InlineData(25, 5)]
+        public void SqrtResultTest(double number, double expectedResult)
+        {
+            //Precondition: A double number number for which to calculate the square root.
+            //Test: Check that the Sqrt method returns the correct square root for different input values.
+            expectedResult = Math.Sqrt(number);
+            double result = calculator.Sqrt(number);
+            Assert.Equal(expectedResult, result);
+        }
+               
+        [Theory]
+        [InlineData(5, true)]
+        [InlineData(-5, false)]
+        [InlineData(0, false)]
+        public void isPositiveReturnsCorrectValueTest(double number, bool expectedValue)
+        {
+            //Precondition: A double number number to check if it is positive, and the expected boolean expectedValue.
+            //Test: Chek that the isPositive method correctly determines whether the given number is positive.
+            bool actualResult = calculator.isPositive(number);
+            Assert.Equal(expectedValue, actualResult);
+        }
+
+        [Theory]
+        [InlineData(5, false)]
```

Comment:
it is better to split negative and positive verifications to different test methods.